### PR TITLE
[Needs Detailed Changelog]: Adding storage account properties to SQL threat detection cmdlets

### PR DESCRIPTION
**Pull request by yoavrubin**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlServerAuditingPolicy.cs [hunk 1]
```diff
@@ -42,12 +42,6 @@ public class SetAzureSqlServerAuditingPolicy : SqlDatabaseServerAuditingCmdletBa
         [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The set of the audit action groups")]
         public AuditActionGroups[] AuditActionGroup { get; set; }
 
-        /// <summary>
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlServerAuditingPolicy.cs]: This is a breaking change.  Please provide some mapping between this parameter and current functionality

Comment (yoavrubin) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlServerAuditingPolicy.cs]: It is not a breaking change. This is a removal of a mistakenly added parameter that is not supported on the server level.

Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlServerAuditingPolicy.cs]: So there is no working code that uses this parameter?  If so, this is fine.

Also, please document the changes in: https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Sql/ChangeLog.md, under 'current rrelease'

Comment (yoavrubin) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlServerAuditingPolicy.cs]: A code that uses this parameter is broken.

File: src/ResourceManager/Sql/ChangeLog.md [hunk 2]
```diff
@@ -19,4 +19,6 @@
 -->
 ## Current Release
 
-## Version 2.3.0
\ No newline at end of file
+## Version 2.3.0
+
+* Added storage properties to threat detection cmdlets
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/ChangeLog.md]: @yoavrubin please move the comment under the "Current Release" section; the "Version 2.3.0" section contains changes made in the previous release, and the "Current Release" section contains changes that will be made for the next Sql release.

**Comment by cormacpayne** [comment]
@markcowl do you approve the changes?

File: src/ResourceManager/Sql/ChangeLog.md [hunk 3]
```diff
@@ -18,7 +18,6 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Added storage properties to threat detection cmdlets
```
Comment (markcowl) [inline src/ResourceManager/Sql/ChangeLog.md]: @yoavrubin Please be specific about which cmdlets are affecetd (please list affected cmdlets)

**Comment by markcowl** [comment]
@yoavrubin The changelog is now published as etadata in your module manifest, so you need to be explicit about what cmdlets are changed, and what functionality is added or removed for each.  Customer can peruse this programmatically through PSGallery cmdlets, or by browsing your module in the gallery.

**Comment by markcowl** [comment]
@yoavrubin Once you have updated the chgnelog with detailed comments, please remove the label in the title and we will merge.

**Comment by cormacpayne** [comment]
Closing this PR since Yaakov is taking over in PR https://github.com/Azure/azure-powershell/pull/3219

**Pull request closed without merging**